### DISCUSSION TITLE
Add variable width and auto content to grid

### DISCUSSION THREAD

**Discussion by leeweisberger** [mantinedev/mantine#2292] [Feature requests]
Implement parity with [this feature in mui grid](https://mui.com/material-ui/react-grid2/#variable-width-content) and [this feature](https://mui.com/material-ui/react-grid2/#auto-layout)

Setting `span` or any breakpoint to `content` makes the grid col take up as much room as its content
Setting `span` or any breakpoint to `auto` makes the grid col take up all additional space.

Happy to send over a PR if this is a feature we'd like to add.

**Comment by rtivital** [comment]
Hi @leeweisberger, it is a great feature, feel free to send a PR to `dev` branch
